### ISSUE TITLE
[Proposal]: Get rid of "Switch" weapon sets and add "Tap" weapon sets instead

### ISSUE THREAD

**Issue by bobbens** [naev/naev#2621]
### Suggestion

I don't think switch weapon sets are really used. It may make more sense to have the primary/secondary keys become their own weapon set. That is, the player would have now a "primary" and "secondary" weapon set, in addition to the current ones, and by assigning weapons there, they could determine their default weapons

Additionally, while hold/toggle weapon sets are nice, they are unable to enable weapons "once" for the duration. For example, for the "Neural Accelerator Interface", the player might want to have it activate for the full duration, independent of how long the weapon was pressed (like the old weapon sets in some cases). The "tap" weapon set would basically activate outfits, but not deactivate them. So they would last until they were manually deactivated, but unlike toggle sets, they would no longer reactivate again.

Thus what the player would likely do is something like:
- primary set -> hold
- secondary set -> hold (or maybe toggle depending on the case)
- neural accelerator interface / blink drives / combat hologram projector and the likes -> tap
- afterburner, other weapons / launchers -> hold
- point defense / fighter bays-> toggle

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bobbens** [comment]
After more discussions with @ThrosturX on discord. It may be possible to go to two modes:
1. Tap -> behaviour depends on outfit, but either taps on, or toggles on.
2. Hold -> actives when held, off when released. Ideal for weapons.

Basically, Tap + Toggle can be outsourced to the outfit themselves. For example, Fighter Bays, Point Defense, or Weapons Ionizer would be toggled, while something like weapons would fire a single volley, or the neural accelerator interface would be turned on until it went off.

Implementation-wise, it could be simplified to two new functions:

```lua
function onhold( p, po, on ) -- same as ontoggle now
end

function ontap( p, po ) -- new, for tap modes
end
```

Since the functionality would be directly to the Lua outfits, they would be able to behave as expected. The default behaviour would be to turn "on" the outfit until it goes off, with exception for non-fighter bay weapons, it would just do a single volley.

**Comment by bobbens** [comment]
I've changed it yet again. Toggle should be the old behaviour. Furthermore, weapons that have no "duration", i.e., never turn off, will be treated as toggle. So now, naturally, weapons and point defense and such are "toggled", while afterburners or astral projections will just stay on until they flip off. Haven't been able to do too much in-depth testing, but it seems to be working well so I've merged into main. Thanks!
